core/db/db.go

```diff
@@ -172,32 +146,23 @@ func getDBPath() string {
 	return dbPath + "db"
 }
 
-func createDBIfDBPathEmpty() error {
-	dbPath := getDBPath()
-	missing, err := dirMissingOrEmpty(dbPath)
-	if err != nil {
-		return err
-	}
-	dbLogger.Debug("Is db path [%s] empty [%t]", dbPath, missing)
-	if missing {
-		err := CreateDB()
-		if err != nil {
-			return nil
-		}
+// Open open underlying rocksdb
+func (openchainDB *OpenchainDB) Open() {
+	openchainDB.mux.Lock()
+	if openchainDB.dbState == opened {
+		openchainDB.mux.Unlock()
+		return
 	}
-	return nil
-}
 
-func openDB() (*OpenchainDB, error) {
-	if isOpen {
-		return openchainDB, nil
-	}
+	defer openchainDB.mux.Unlock()
 
 	dbPath := getDBPath()
+	mkDirIfNotExist(dbPath)
+
 	opts := gorocksdb.NewDefaultOptions()
 	defer opts.Destroy()
 
-	opts.SetCreateIfMissing(false)
+	opts.SetCreateIfMissing(true)
```

Comment:
@GrapeBaBa I think that you are discounting `opts.SetCreateIfMissing(false)` in the openDB. It will raise an error in the case directory has some non-relevant contents.